packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -4844,19 +4878,24 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
       _debugLocked = true;
       return true;
     }());
-    final _RouteEntry entry = _history.lastWhere(_RouteEntry.isPresentPredicate);
-    if (entry.hasPage) {
-      if (widget.onPopPage!(entry.route, result))
+    final int index = _history.lastIndexWhere(_RouteEntry.isPresentPredicate);
+    final _RouteEntry entry = _history[index];
+    if (entry.hasPage && widget.onPopPage!(entry.route, result)) {
```

Comment:
if the entry has a page and widget.onPopPage denied the pop, we shouldn't be executing the else branch and call pop on the entry?

Reply:
nice catch